staging/src/k8s.io/kubectl/pkg/drain/filters.go

```diff
@@ -187,13 +187,15 @@ func (d *Helper) daemonSetFilter(pod corev1.Pod) PodDeleteStatus {
 		return MakePodDeleteStatusOkay()
 	}
 
-	if _, err := d.Client.AppsV1().DaemonSets(pod.Namespace).Get(context.TODO(), controllerRef.Name, metav1.GetOptions{}); err != nil {
-		// remove orphaned pods with a warning if --force is used
-		if apierrors.IsNotFound(err) && d.Force {
-			return MakePodDeleteStatusWithWarning(true, err.Error())
+	// No need to check if the pod is orphaned when `--force=false`
+	if d.Force {
+		if _, err := d.Client.AppsV1().DaemonSets(pod.Namespace).Get(context.TODO(), controllerRef.Name, metav1.GetOptions{}); err != nil {
```

Comment:
Does that mean https://github.com/kubernetes/kubernetes/blob/587f4f04cc5fc18f4e85ed6a4a06bbf1bfee0496/staging/src/k8s.io/kubectl/pkg/drain/filters.go#L199-L203 does not care about this operation and this Get operation is only done for `Force` flag?

Reply:
Yes. Only if Force is true, the Get operation will help to check the orphaned pods for daemonset.